embed/serve.go

```diff
@@ -207,6 +207,10 @@ func grpcHandlerFunc(grpcServer *grpc.Server, otherHandler http.Handler) http.Ha
 	}
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		if r.ProtoMajor == 2 && strings.Contains(r.Header.Get("Content-Type"), "application/grpc") {
+			// gRPC-gateway should not include CN in request
+			if gw := r.Header.Get("Grpcgateway-Accept"); gw != "" {
+				r.TLS.PeerCertificates[0].Subject.CommonName = ""
```

Comment:
Is this meaningful? The problem is CN from a client isn't propagated to servers.
Also modifying a given TLS information wouldn't be a good practice, I believe.

Reply:
Yes this is meaningful, the request in this case is isolated from gRPC-gateway proxy after client request because of the header details we know this. If the etcd client server cert has a CN then it will always be forwarded to server by proxy request which can be dangerous. We are not making a change to client request here. But instead not forwarding TLS information which was never made by the client to begin with.

Reply:
I see. I misunderstood the code, thanks.

But anyway I think modifying fields of TLS is not a good practice because it makes an object of `tls.ConnectionState` inconsistent. e.g. `ConnectionState.VerifiedChains.Raw*` will become different from other parsed fields (I think the modification shouldn't be allowed by the library. I'm wondering why this is allowed).

Even if we do this, I believe this should be for defensive purpose. Returning an error in earlier step like this: https://github.com/etcd-io/etcd/pull/10343/commits/0c68a9f4ff34a03ecc715787e1c845a3e063e349 is enough for preventing the problem and helpful for diagnosing. From the perspective of users, current behavior of this PR is confusing (TLS connection can be established but CN in cert is ignored). Because of this reason, I think adding a warn level log should be added for letting users know about this modification.

Reply:
@mitake I agree WRT TLS Info and have adjusted my approach.